proj/src/main/webapp/chat.js

```diff
@@ -576,6 +580,17 @@ function createMessageWithTemplate(key, messageObj) {
     timestamp.dataMilli = messageObj.timestamp;
     timestamp.innerText = new Date(messageObj.timestamp).toLocaleString();
 
+    // add pfp
+    const pfpRef = firebase.database().ref('/users/'+messageObj.senderUID+'/photo');
+    pfpRef.once('value', function(snap) {
+        const url = snap.val() || "gs://arringtonh-step-2020-d.appspot.com/default.png";
+        storageRef = firebase.storage().refFromURL(url);
+        storageRef.getDownloadURL().then(function(src) {
+            message.querySelector('#pfp').src = src;
+        })
+    })
```

Comment:
The image might be something you want to cache in the browser and reference so that if the same user sent 10 messages, you won't download the same image 10 times. But at the same time you'll want to clear the cache whenever the chat changes.